sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -2007,4 +2008,78 @@ class RegexUnsupportedException(message: String, index: Option[Int])
       case _ => message
     }
   }
-}
\ No newline at end of file
+}
+
+sealed trait RegexOptimizationType
+object RegexOptimizationType {
+  case class StartsWith(literal: String) extends RegexOptimizationType
+  case class Contains(literal: String) extends RegexOptimizationType
+  case object NoOptimization extends RegexOptimizationType
+}
+
+object RegexRewriteUtils {
+  private def isliteralString(astLs: collection.Seq[RegexAST]): Boolean = {
+    astLs.forall {
+      case RegexChar('^') | RegexChar('$') | RegexChar('.') => false
+      case RegexChar(_) => true
+      case _ => false
+    }
+  }
+
+  private def isWildcard(ast: RegexAST): Boolean = {
+    ast match {
+      case RegexRepetition(RegexChar('.'), SimpleQuantifier('*')) => true
+      case RegexSequence(parts) if parts.forall(isWildcard) => true
+      case RegexGroup(_, term, _) if isWildcard(term) => true
+      case _ => false
+    }
+  }
+
+  private def stripLeadingWildcards(astLs: collection.Seq[RegexAST]): 
+      collection.Seq[RegexAST] = astLs match {
+    case (RegexChar('^') | RegexEscaped('A')) :: tail  =>
+      tail.dropWhile(isWildcard)
+    case _ => astLs.dropWhile(isWildcard)
+  }
+
+  private def stripTailingWildcards(astLs: collection.Seq[RegexAST]): 
+      collection.Seq[RegexAST] = {
+    astLs.reverse.dropWhile(isWildcard).reverse
+  }
+
+  private def RegexCharsToString(chars: collection.Seq[RegexAST]): String = {
+    chars.map {
+      case RegexChar(ch) => ch
+      case _ => throw new IllegalArgumentException("Invalid character")
+    }.mkString
+  }
+
+  def matchSimplePattern(ast: RegexAST): RegexOptimizationType = {
```

Comment:
Could we get a few scaladocs for the API, so it is clear when and how to use it?